plugins/uploader-resources/src/components/FileUploadStatusPopup.svelte

```diff
@@ -12,115 +12,80 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 -->
+
 <script lang="ts">
   import { getEmbeddedLabel } from '@hcengineering/platform'
-  import {
-    Button,
-    IconClose,
-    Label,
-    ProgressCircle,
-    Scroller,
-    humanReadableFileSize as filesize,
-    tooltip
-  } from '@hcengineering/ui'
+  import { Button, IconClose, Label, ProgressCircle, Scroller, tooltip } from '@hcengineering/ui'
   import { ObjectPresenter } from '@hcengineering/view-resources'
-  import type { UppyFile, State as UppyState } from '@uppy/core'
-  import { createEventDispatcher, onDestroy, onMount } from 'svelte'
+  import { createEventDispatcher, onMount } from 'svelte'
 
   import IconCompleted from './icons/Completed.svelte'
   import IconError from './icons/Error.svelte'
   import IconRetry from './icons/Retry.svelte'
 
   import uploader from '../plugin'
-  import { type FileUpload } from '../store'
+  import { uploads, type Upload, type FileUpload } from '../store'
 
-  export let upload: FileUpload
+  export let uploadId: string
 
   const dispatch = createEventDispatcher()
+  let upload: Upload | undefined = undefined
 
-  let state: UppyState<any, any> = upload.uppy.getState()
-
-  $: state = upload.uppy.getState()
-  $: files = Object.values(state.files)
-  $: capabilities = state.capabilities ?? {}
-  $: individualCancellation = 'individualCancellation' in capabilities && capabilities.individualCancellation
-
-  function updateState (): void {
-    state = upload.uppy.getState()
-  }
-
-  function handleComplete (): void {
-    if (upload.uppy.getState().error === undefined) {
+  function updateState (state: Map<string, Upload>): void {
+    upload = state.get(uploadId)
+    if (upload === undefined) {
       dispatch('close')
+      return
     }
-  }
-
-  function handleFileRemoved (): void {
-    state = upload.uppy.getState()
-    const files = upload.uppy.getFiles()
-    if (files.length === 0) {
+    if (upload.progress === 100 && upload.error === undefined) {
+      dispatch('close')
+    }
+    if (upload.files.size === 0) {
       dispatch('close')
     }
   }
 
   function handleCancelAll (): void {
-    upload.uppy.cancelAll()
+    upload?.files.forEach((element) => {
+      if (element.cancel !== undefined) {
+        element.cancel()
```

Comment:
nit: here and in other places can be shortened: `element.cancel?.()`

Reply:
thanks, fixed.